src/lib/io/sol-iio.c

```diff
@@ -963,9 +1035,26 @@ sol_iio_close(struct sol_iio_device *device)
 {
     int i;
     struct sol_iio_channel *channel;
+    struct sol_list *it = NULL, *it_next = NULL;
+    struct open_list *entry = NULL;
 
     SOL_NULL_CHECK(device);
 
+    SOL_LIST_FOREACH_SAFE (&iio_open_list, it, it_next) {
+        entry = SOL_LIST_GET_CONTAINER(it, struct open_list, list);
+        if (device->device_id == entry->device->device_id) {
+            entry->refcount--;
+            if (entry->refcount == 0) {
+                SOL_DBG("Close %d", device->device_id);
+                sol_list_remove(it);
+                free(entry);
+            } else {
+                SOL_DBG("cannot close %d", device->device_id);
```

Comment:
I don't think this message is useful. Either delete it or write something like 'device X still have N references'.